File: my-solutions/0125-valid-palindrome/solution.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        transform(s.begin(), s.end(), s.begin(), [](unsigned char c){ return tolower(c); });
        s.erase(remove_if(s.begin(), s.end(), [](char c){ return !std::isalnum(c); }), s.end());

        string rev = s;
        reverse(rev.begin(), rev.end());

        return s == rev;
    }
};

File: my-solutions/0125-valid-palindrome/solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        clean_str = re.sub(r'[^a-zA-Z0-9]', '', s).lower()
        if clean_str == "":
            return True
        
        start = 0
        end = len(clean_str) - 1
        while start < end:
            if clean_str[start] != clean_str[end]:
                return False
            start += 1
            end -= 1

        return True
        
